### ISSUE TITLE
Proposal: Improve desktop-gui error messages

### ISSUE THREAD

**Issue by chrisbreiding** [cypress-io/cypress#2955]
Currently, errors displayed in the desktop-gui could use some improvement. This is what one currently looks like:

![screen shot 2018-12-17 at 12 15 22 pm](https://user-images.githubusercontent.com/1157043/50103361-77cc3e00-01f5-11e9-9b83-a99ce6eaad93.png)

Make the following improvements:

- Add markdown support
- Improve the general styling (left-align text, etc)
- Collapse any stack trace that's part of the error, with a button to expand

**Comment by bahmutov** [comment]
another improvement: allow multiple links to our docs / blog posts / recipes that explain the problem and provide solutions or alternatives

**Comment by lilaconlee** [comment]
Remaining work: 
- [ ] Fix failing e2e snapshot test
- [ ] Add support for opening links in browser not electron

A couple screenshots of collapsing error details: 
<img width="799" alt="screen shot 2018-12-26 at 4 47 36 pm" src="https://user-images.githubusercontent.com/7514489/50458036-2ecf6480-092e-11e9-9a0d-1ae4b2f15a16.png">
<img width="939" alt="screen shot 2018-12-20 at 10 21 07 am" src="https://user-images.githubusercontent.com/7514489/50458039-30009180-092e-11e9-82d2-702e67dc0a38.png">

**Comment by jennifer-shehane** [comment]
Looks so much better 😍

**Comment by lilaconlee** [comment]
Yeah, it's scrollable but you can't tell until you try and scroll it. I'm down to make it always visible if we'd prefer.

**Comment by lilaconlee** [comment]
Updated warning styling as well (believe this is the only warning):
<img width="1126" alt="screen shot 2019-01-14 at 11 32 11 am" src="https://user-images.githubusercontent.com/7514489/51143553-ec919880-17f2-11e9-9b87-e353b56a885a.png">

**Comment by lilaconlee** [comment]
After review with Chris and Brian, remaining work:
- [ ] Revert snapshot changes
- [ ] Update `run.coffee` to log `err.message` and `err.details` 
- [ ] Confirm snapshot failures are limited to the expected ones (async plugins failure)
- [ ] Update snapshot

**Comment by lilaconlee** [comment]
Diff from updating `run.coffee`, think this is what we want so I'm going to go ahead and update the snapshots: 
<img width="811" alt="screen shot 2019-01-17 at 6 05 32 am" src="https://user-images.githubusercontent.com/7514489/51331506-006d1280-1a1e-11e9-97dc-001dba9e03f0.png">

**Comment by brian-mann** [comment]
@lilaconlee 	in order for github to show you the number of outstanding tasks for issues the tasks must be listed in the first comment. same goes for PR's too.
